### PR TITLE
Save node list on close

### PR DESCRIPTION
**Pull request by lukechampine**
The node list wasn't being saved after removing nodes, so the gateway now calls `g.save` in `g.Close`. Saving inside `g.removeNode` is another option, but I wanted to avoid saving repeatedly when removing many nodes.

TODO: the gateway should ask for more nodes if its node list is getting too sparse.

### CODE REVIEW

File: modules/gateway/peers_test.go [hunk 1]
```diff
@@ -28,20 +28,40 @@ func TestListen(t *testing.T) {
...
-	// "compliant" connect
+	// compliant connect with old version
 	conn, err := net.Dial("tcp", string(g.Address()))
 	if err != nil {
 		t.Fatal("dial failed:", err)
 	}
 	addr := modules.NetAddress(conn.LocalAddr().String())
 	// send version
-	if err := encoding.WriteObject(conn, build.Version); err != nil {
+	if err := encoding.WriteObject(conn, "0.1"); err != nil {
```
Comment (DavidVorick) [inline modules/gateway/peers_test.go]: 0.1?

Comment (lukechampine) [inline modules/gateway/peers_test.go]: it just needs be a version old enough to be rejected.

**Comment by DavidVorick** [comment]
build failed

**Comment by DavidVorick** [comment]
jk

**Pull request merged**